File: sdks/dotnet/sulfone-helium/Domain/Extension/Input.cs
using sulfone_helium.Domain.Core;

namespace sulfone_helium.Domain.Extension;

// Server Inputs (Exposed as requests)
public record ExtensionAnswerInput(
    IAnswer[] Answers,
    Dictionary<string, string>[] DeterministicState,
    IAnswer[] PrevAnswers,
    Cyan PrevCyan,
    Dictionary<string, IEnumerable<IAnswer>> PrevExtensionAnswers,
    Dictionary<string, Cyan> PrevExtensionCyans
);

public record ExtensionValidateInput(
    IAnswer[] Answers,
    Dictionary<string, string>[] DeterministicState,
    IAnswer[] PrevAnswers,
    Cyan PrevCyan,
    Dictionary<string, IEnumerable<IAnswer>> PrevExtensionAnswers,
    Dictionary<string, Cyan> PrevExtensionCyans,
    string Validate
);

File: sdks/node/src/domain/extension/input.ts
import type { Answer } from "../core/answer.js";
import type { Cyan } from "../core/cyan.js";

interface ExtensionAnswerInput {
  answers: Answer[];
  deterministicState: Record<string, string>[];
  prevAnswers: Answer[];
  prevCyan: Cyan;
  prevExtensionAnswers: Record<string, Answer[]>;
  prevExtensionCyans: Record<string, Cyan>;
}

interface ExtensionValidateInput {
  answers: Answer[];
  deterministicState: Record<string, string>[];
  prevAnswers: Answer[];
  prevCyan: Cyan;
  prevExtensionAnswers: Record<string, Answer[]>;
  prevExtensionCyans: Record<string, Cyan>;
  validate: string;
}

// export all
export type { ExtensionAnswerInput, ExtensionValidateInput };
